src/Bicep.Cli/Services/ValidationService.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Text;
+using Bicep.Core.Emit;
+using Bicep.Core.Semantics;
+using Microsoft.WindowsAzure.ResourceStack.Common.Collections;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Bicep.Cli.Services
+{
+    public class ValidationService
+    {
+        public static void Validate(ImmutableArray<TestSymbol> testDeclarations)
+        {
+            var templateOutputBuffer = new StringBuilder();
+            using var textWriter = new StringWriter(templateOutputBuffer);
```

Comment:
No need to create a `StringBuilder` manually. The default `StringWriter` constructor will create one.
```suggestion
            using var textWriter = new StringWriter();
```